packages/web-components/fast-foundation/src/date-picker/date-picker.template.ts

```diff
@@ -0,0 +1,254 @@
+                part="picker-cell"
+                grid-column="${(x, c) => c.index + 1}"
+                @click="${x => x.action()}"
+                @keydown="${(x, c) => x.action()}"
+              >
+                ${x => x.text}
+              </${cell}>
+            `,
+                { positioning: true }
+            )}
+          </${row}>
+        `
+        )}
+      </${grid}>
+      ${reset}
+    </div>
+  `;
+};
+
+/**
+ *
+ * @param context - The date-picker class context
+ * @param definition - Date-picker options
+ * @returns - A ViewTemplate
+ * @public
+ */
+export const datePickerTemplate: (
+    Context: ElementDefinitionContext,
+    definition: DatePickerOptions
+) => ViewTemplate<DatePicker> = (
+    context: ElementDefinitionContext,
+    definition: DatePickerOptions
+) => {
+    const textField = context.tagFor(TextField);
+    const anchoredRegion = context.tagFor(AnchoredRegion);
+    const button = context.tagFor(Button);
+    const calendar = context.tagFor(Calendar);
+    const resetButton = html`<${button} class="reset-text" part="reset-text" @click="${x =>
+        x.resetCalendar()}">${x => x.resetText}</${button}>`;
+
+    return html`
+    <template
+        @mouseover="${x => (x.overFlyout = true)}"
+        @mouseout="${x => (x.overFlyout = false)}"
+    >
+        <${textField}
+            ${ref("textField")}
+            name=${x => x.name}
+            @click="${x => (!x.readonly ? x.toggleFlyout(true) : () => {})}"
+            ?readonly="${x => x.readonly}"
+            placeholder="${x => x.placeholder}"
+            @blur="${x => x.handleBlur()}"
+            @focus="${x => x.handleFocus()}"
+            @keyup="${(x, c) => x.handleKeyup(c.event)}"
+        >
+            <div slot="end">&#128197;</div>
+        </${textField}>
+        ${when(
+            x => !x.readonly,
+            html`
+            <${anchoredRegion}
+                class="flyout ${x => (x.flyoutOpen ? "show" : "")}"
+                part="flyout"
+                :anchorElement="${x => x.textField}"
+                vertical-positioning-mode="dynamic"
+                vertical-default-position="bottom"
+                horizontal-positioning-mode="dynamic"
+                horizontal-inset="true"
+                horizontal-default-position="start"
+            >
+                ${when(
+                    x => x.type === "datetime-local" || x.type === "time",
+                    html`
+                        ${x => timePickerTemplate(context, x.getTimes())}
+                    `
+                )}
+                ${when(
+                    x => x.showCalendar,
+                    html`
+                    <${calendar}
+                        selected-dates="${x => `${x.date || ""}`}"
+                        class="calendar"
+                        @dateselected="${(x, c) =>
+                            x.handleDateClicked((c.event as MouseEvent).detail)}"
```

Comment:
Why parse the event detail in the template and not in the class handler?  It seems like an unnecessary template dependency should the handler become interested in other event properties.

Reply:
I originally was passing detail because the TypeScript definition for `MouseEvent.detail: number` was throwing errors. Trying to cast event.detail to `{day: number; month: number; year: number;}` was throwing a TypeScript error that it suspected it was an error in typing. I've had to cast it to `event.detail as any` to keep TypeScript happy.

Reply:
Not what I was asking, rather why not just pass `(c.event as MouseEvent)` to the handler and grab the `detail` from the event in the class vs the template?  I don't think it makes a difference perf wise and it makes it easier to have the handler look at other aspects of the event should it become necessary without also requiring a template change.

Reply:
No, I totally get what you're saying. And I did pass c.event originally. But TypeScript has MouseEvent.detail typed as a number. I'm passing an object with day, month, year. It throws an error because it's typed as a number. When I try to cast MouseEvent.detail to {day, month, year} in date-picker.ts, it throws an error thinking that I made a mistake. The only way to access event.detail.day, event.detail.month and event.detail.year is to cast event.detail as any. This felt a little dirty, so I moved it to the template and was going to deal with it later. But hadn't. I've updated the code per your comments. You can see it code now.

```
    public handleDateClicked(event: MouseEvent): void {
        const { day, month, year } = event.detail as any;
        this.setValue({ day, month, year });
    }
```